src/modal/modal.ts

```diff
@@ -21,8 +21,11 @@ export class NgbModal {
    * NgbActiveModal class to close / dismiss modals from "inside" of a component.
    */
   open(content: any, options: NgbModalOptions = {}): NgbModalRef {
+    const injector = options.injector || this._injector;
+    const cfr = injector.get<ComponentFactoryResolver>(ComponentFactoryResolver as any, this._moduleCFR);
     const combinedOptions = Object.assign({}, this._config, options);
-    return this._modalStack.open(this._moduleCFR, this._injector, content, combinedOptions);
+
+    return this._modalStack.open(cfr, this._injector, content, combinedOptions);
```

Comment:
I am not starting a proper review here, but just throwing one comment: shouldn't `this._injector` be `injector` instead as the second argument ?

Reply:
Yes and no!
With he current implementation of the ```ModalStack``` it does not make any difference. The ```ModalStack``` uses the injector from the ```options``` if given - if not it uses the one passed as 2nd argument. I decided to keep this as the ```ModalStack``` then still has access to both injectors (if they might be different). Although I do not see at the moment a use cases where the ```ModalStack``` needs both of them. My intention was not to change too much in this bugfix-PR. Of course I can change this if desired.